### PR TITLE
feat(args): add offline flag

### LINKED ISSUE #1095: Add offline/local version of --bumped-version
### Is there an existing issue or pull request for this?

- [x] I have searched the existing issues and pull requests

### Feature description

This feature would enable calculation of the bumped version without requiring an authentication token (e.g. GITLAB_TOKEN) and without need to access the remote. It purely consults the local git history and tags to make its calculation.

### Desired solution

It would be nice if --bumped-version could be local/offline by default, but alternatively, one could add a flag that sets that preference, e.g. --offline or --local.

### Alternatives considered

I don't think there is an alternative, other than trying a different utility, but then we would have to define our "bump rules" for two different tools and keep them in sync.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by barskern**
## Description

Add a `--offline` flag to enable running e.g. `--bumped-version` without contacting any external services, even if they are configured. Closes #1095.

## Motivation and Context

See #1095 .

## How Has This Been Tested?

Local unit and integration tests, including building and running the binary in my local repo.

**NOTE!** Three unrelated tests were failing when I started my implementation:

```
repo::test::get_latest_tag
repo::test::git_tags
repo::test::resolves_existing_tag_with_name_and_message
```

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: git-cliff/src/args.rs [hunk 1]
```diff
@@ -359,6 +359,9 @@ pub struct Opt {
     /// Load TLS certificates from the native certificate store.
     #[arg(long, help_heading = Some("FLAGS"), hide = !cfg!(feature = "remote"))]
     pub use_native_tls: bool,
+    /// Use only local git history in actoins
```
Comment (orhun) [inline git-cliff/src/args.rs]:
```suggestion
    /// Use only local git history in actions
```

File: git-cliff-core/src/config.rs [hunk 2]
```diff
@@ -177,6 +177,11 @@ mod serde_pattern {
 /// Remote configuration.
 #[derive(Default, Debug, Clone, Serialize, Deserialize)]
 pub struct RemoteConfig {
+    /// Run generation fully offline
+    // NOTE! Intended to be set from the CLI args, as having it in
+    // the config file doesn't really makes sense IMO.
```
Comment (orhun) [inline git-cliff-core/src/config.rs]: I think we can also support it the configuration file :)

```suggestion
```

**Comment by barskern** [comment]
> Looks good! Also we need to update the documentation/website about this new flag :)

Thank you for your patience here! I've added your suggested changes and a small section to the website. I was also looking for a way to test it, but in the way the `remote` module is implemented for the moment I wasn't able to find a clean way to do it. It seems like the remote implementations are not really tested for the moment?

**Comment by orhun** [comment]
Check out the [test fixtures here](https://github.com/orhun/git-cliff/tree/main/.github/fixtures). I believe we can copy one of the GitHub integration ones, run it with `--offline` and compare the result (`expected.md`).

**Comment by barskern** [comment]
It was good that you requested me to create a test! My ignorance thought this was a trivial change, but making the test made me catch a edge-case. Thank you!

**Comment by W1M0R** [comment]
Thank you for this feature @barskern and @orhun!
